spiketoolkit/comparison/groundtruthcomparison.py

```diff
@@ -0,0 +1,291 @@
+
+    def print_performance(self, method='by_spiketrain'):
+        """
+        Print performance with the selected method
+        """
+        if method == 'by_spiketrain':
+            perf = self.get_performance(method=method, output='pandas')
+            perf = perf * 100
+            #~ print(perf)
+            d = {k: perf[k].tolist() for k in perf.columns}
+            txt = _template_txt_performance.format(method=method, **d)
+            print(txt)
+
+        elif method == 'pooled_with_sum':
+            perf = self.get_performance(method=method, output='pandas')
+            perf = perf * 100
+            txt = _template_txt_performance.format(method=method, **perf.to_dict())
+            print(txt)
+
+        elif method == 'pooled_with_average':
+            perf = self.get_performance(method=method, output='pandas')
+            perf = perf * 100
+            txt = _template_txt_performance.format(method=method, **perf.to_dict())
+            print(txt)
+    
+    def print_summary(self):
+        """
+        Print a global performance summary that depend on the context:
+          * exhaustive= True/False
+          * how many gt units (one or several)
+        
+        This summary mix several performance metrics.
+        """
+        raise(NotImplementedError)
+    
+    def get_well_detected_units(self, tp_thresh=0.95, accuracy_thresh=None,
+                cl_thresh=None, fp_thresh=None):
+        """
+        Get the units in GT that are well detected with a comninaison a treshold level
+        on some columns (tp_rate, accuracy_rate, cl_rate, fp_rate):
+        
+        If several thresh are given the the intersect of selection is kept.
+        
+        Parameters
+        ----------
+        tp_thresh:
+            Threshold tp_rate score above a units is selected
+        
+        accuracy_thresh:
+            Threshold accuracy score above a units is selected
+        
+        cl_thresh:
+            Threshold cl_rate score under a units is selected
+        
+        fp_thresh:
+            Threshold fp_rate score under a units is selected
+        
+        """
+        perf = self.get_performance(method='by_spiketrain')
+        keep = perf['tp_rate']>=0
+        
+        if tp_thresh is not None:
+            keep = keep & (perf['tp_rate'] >= tp_thresh)
+        
+        if accuracy_thresh is not None:
+            keep = keep & (perf['accuracy'] >= accuracy_thresh)
+        
+        if cl_thresh is not None:
+            keep = keep & (perf['cl_rate'] <= cl_thresh)
+        
+        if fp_thresh is not None:
+            keep = keep & (perf['fp_rate'] <= fp_thresh)
+        
+        return perf[keep].index.tolist()
+    
+    def count_well_detected_units(self, **kargs):
+        """
+        Count how many well detected units.
+        Kargs are the same as get_well_detected_units.
+        """
+        return len(self.get_well_detected_units())
+    
+    def get_fake_units_in_other(self):
```

Comment:
I would call it `unmatched_units`

Reply:
I also don't like "fake_units".

But here we need a clear semantic. The pathology we are tracking here is that some sorter have many more units they should have and worse: sometime a gt units is matched to several units in sorter. The best one with a good TP score, the others matched must counted and should penalize the final benchmark. This is the goal of this function.

The semantic "unmatched_unit" in my opinion is more related to a not detected units, in short unmatched in the GT (the other way).

Could a good english speaker could make other other proposal ?
@colehurwitz31 @mhhennig ?

Reply:
then they should be false_detected_units?

Reply:
Quite good. lets wait other proposal.
The most important will be the documentation coming with this.

Reply:
What about “false_units”? Or is that too vague?

Reply:
Or maybe, “extra_units”

Reply:
Or maybe “fake_units”

Reply:
héhé fake_units was the proposal but Alessio do not like it. I am also not very happy with it.
I like when samtic about metric are cristal clear.